plugin/testdata/override-ngx-re-test.lua

```diff
@@ -0,0 +1,30 @@
+local typedefs = require "kong.db.schema.typedefs"
+local re_find = ngx.re.find
+local re_match = ngx.re.match
+
+return {
+    name = "override-ngx-re-test",
+    fields = {
+        { config = {
+            type = "record",
+            fields = {
+                { findstr = { type = "string" }, },
+                { matchstr = { type = "string" }, },
+            },
+            custom_validator = function(config)
+                if type(config.findstr) == "string" then
+                    if not re_find(config.findstr, "([0-9]+)") then
+                        return false, "find fail"
+                    end
+                end
+                if type(config.matchstr) == "string" then
+                    if not re_match(config.matchstr, "^([a-z]+)") then
+                        return false, "match fail"
+                    end
+                end
+                return true
+            end,
+        },
+        },
+    },
+}
```

Comment:
- nit: Two spaces for indents of Lua files (not four)
- fields could have two regexes and a subject to simplify
  - custom validator should utilize these new regex fields for extending tests on the go side
- nit: no need to create `re_find` and `re_match`, just use ngx.re.find|match directly
- nit: just call plugin name `ngx-re-test`

A better schema for validation and testing with varying values could be (untested schema)

```
local typedefs = require "kong.db.schema.typedefs"

return {
  name = "ngx-re-test",
  fields = {
    { config = {
      type = "record",
      fields = {
        { find_regex = { type = "string", is_regex = true }, },
        { match_regex = { type = "string", is_regex = true }, },
        { subject = { type = "string", required = true }, },
      },
      custom_validator = function(config)
        if config.find_regex ~= ngx.null and #config.find_regex > 0 then
          local from = ngx.re.find(config.subject, config.find_regex)
          if not from then
            return false, "failed to find subject " .. config.subject ..
              " using regex " .. config.find_regex
          end
        end
        if config.match_regex ~= ngx.null and #config.match_regex > 0 then
          local m = ngx.re.match(config.subject, config.match_regex)
          if not m then
            return false, "failed to match subject " .. config.subject ..
              " using regex " .. config.match_regex
          end
        end
        return true
      end,
      },
    },
  },
  entity_checks = {
    { at_least_one_of = { "config.find_regex", "config.match_regex" } },
  },
}

```

Reply:
I looked at 19 regex/target strings used in `match` calls across several lua files. I've added 4 unit tests that use existing literals. I came across several regexes that seemingly have a malformed input. Others did not have existing tests or adequate documentation for me to efficiently derive a potential matching string to test against.

Complete list:
```
tested:
* uuid regex ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$
* header name ^[a-zA-Z0-9-_]+$
* authorization_header: ([^:]+):(.*)
* /directory$

malformed?
* vault: [[([^/]+)$]] mismatched )
* vault: (/[^/]+) unescaped / 
* casandra: [==[[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\:\$\ dangling backslash
* get response type: [[^\s*q=([0-9]*[\.][0-9]+)\s*$]] unmatched closing peren )
* parse_directive_header: [[^\s*([^=]+)(?:=(.+))?]] unmatched closing peren )

unsure of input:
* decode array arg: [[(.+)\[(\d*)\]$]]
* decode arg [[^\.+|\.$]]
* get response type [[^\s*(\S+\/\S+)\s*$]]
* no Route matched with those values
* /$
* [[\.well-known/]] .. acme-challenge .. /(.+)
* ^([" ..  [[\x21-\x2B\x2D\x2E\x30-\x7E\x80-\xF4]] .. "]+(?:,|$))+$
* ^([" ..  [[\x21-\x2B\x2D\x2E\x30-\x7E\x80-\xF4]] .. "]+(?:/|$))+$
* (VALUE\)]==]
* specific string "any string"
```